### DISCUSSION TITLE
deepflow Helm Chart中自带的Grafana中的初始化容器有什么特殊作用？

### DISCUSSION THREAD

**Discussion by gzOne2Free** [deepflowio/deepflow#3771] [Q&A] [answered]
我在公司的内网环境部署了一套deepflow PoC环境，但是Grafana容器一直在CrashLoopBackOff，我们用Grafana原厂的Chart是没有问题的。检查了一下发现比官方的多了2个InitContainers，分别是deepflowio-init-grafana:v6.3.3和deepflowio-init-grafana-ds-dh:latest。怀疑是这2个容器需要联Internet（内网访问不了Internet，需要开墙）导致容器启动失败。
想咨询一下如果这2个容器会访问Internet，是访问哪些网址呢？我们需要申请开通防火墙。
又或者这2个初始化容器的作用是什么？能不能在现有的Grafana上手工操作实现（例如目的是安装插件、导入Dashboard template）？

**Comment by Nick-0314** [comment, marked as answer]
不需要访问外网，用于初始化datasource、dashboard 、plugin
